sympy/assumptions/ask.py

```diff
@@ -68,6 +68,22 @@ def antihermitian(self):
         # TODO: Add examples
         return Predicate('antihermitian')
 
+    @predicate_memo
+    def idempotent(self):
+        """
+        Idempotent predicate.
+
+        ``Q.idempotent(x)`` is true iff ``x^n=x`` for non-negative ``x``. 
```

Comment:
Should this be non-negative n rather than x?

Reply:
Indeed, corrected.